packages/flutter_test/lib/src/mock_event_channel.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// A mock stream handler for an [EventChannel] that mimics the native
+/// StreamHandler API.
+///
+/// {@template flutter.flutter_test.MockStreamHandler}
+/// The [onListen] callback is provided a [MockStreamHandlerEventSink] with
+/// the following API:
+/// - [MockStreamHandlerEventSink.success] sends a success event.
+/// - [MockStreamHandlerEventSink.error] sends an error event.
+/// - [MockStreamHandlerEventSink.endOfStream] sends an end of stream event.
+/// {@endtemplate}
+abstract class MockStreamHandler {
+  /// Create a new [MockStreamHandler].
+  MockStreamHandler();
+
+  /// Create a new [InlineMockStreamHandler] with the given [onListen] and
+  /// [onCancel] handlers.
+  factory MockStreamHandler.inline({
+    required MockStreamHandlerOnListenCallback onListen,
+    MockStreamHandlerOnCancelCallback? onCancel,
+  }) => InlineMockStreamHandler._(onListen: onListen, onCancel: onCancel);
+
+  /// Handler for the listen event.
+  void onListen(Object? arguments, MockStreamHandlerEventSink events);
+
+  /// Handler for the cancel event.
+  void onCancel(Object? arguments);
+}
+
+/// Typedef for the [InlineMockStreamHandler.onListen] callback.
+typedef MockStreamHandlerOnListenCallback = void Function(Object? arguments, MockStreamHandlerEventSink events);
+
+/// Typedef for the [InlineMockStreamHandler.onCancel] callback.
+typedef MockStreamHandlerOnCancelCallback = void Function(Object? arguments);
+
+/// Convenience class for creating a [MockStreamHandler] inline.
+///
+/// {@macro flutter.flutter_test.MockStreamHandler}
+class InlineMockStreamHandler extends MockStreamHandler {
```

Comment:
Can this class just be private? Since the only way to construct this is via `MockStreamHandler.inline`, what's the need for having this class public?

Reply:
Making this private makes it impossible to assign it to a variable if the user's analysis options force verbose type specification

Reply:
You'd just type the variable as `MockStreamHandler`, which is what I would expect for a class constructed with `MockStreamHandler.inline(...)`. If you wanted to assign it to a variable of `InlineMockStreamHandler` you'd have to type-cast it, which seems very odd to do.

Reply:
Good point. I made it private.